docs/concepts/11-versions.md

```diff
@@ -0,0 +1,609 @@
+- Not free-form - must select from published versions
+- Development releases are **hidden** from this interface
+
+**Generated Release Channels:**
+- `~1`, `~1.1`, `~2`, `~2.1` (auto-generated tilde patterns for each possible)
+- `latest` - Newest stable release
+- When you set a version constraint, `currentVersion` gets updated to show what will deploy
+
+**Development Releases Toggle:**
+- Available when selecting any release channel (`~1`, `~1.1`, `latest`, etc.)
+- When enabled, the channel will also match and automatically deploy development releases
+- Example: `~1.0` with development releases enabled will automatically deploy `1.0.5-dev.*` versions
+- **Perfect for testing environments**: Automatically validates new IaC changes
+
+#### Version Constraints
+
+**Upgrade-Only Policy:**
+- ✅ Can upgrade: `1.0.0` → `1.1.0` → `2.0.0`
+- ❌ Cannot downgrade: `2.0.0` → `1.1.0`
+- ✅ Can rollback: Restore previous configuration with its original version
+
+**Configuration Snapshots:**
+Each version change creates a configuration snapshot, enabling safe rollbacks to previous states while maintaining the upgrade-only constraint for new deployments.
+
+## Upgrade Strategies
+
+### Feature Upgrades  
+Use minor channels for patch updates:
+```
+~1.1    # Patch updates only (1.1.0 → 1.1.5)
+```
+
+### Major Upgrades
+Use major channels for latest features:
+```
+~1      # All updates within major 1 (1.0.0 → 1.9.5)
+```
+
+## Best Practices
+
+### For Bundle Developers
+
+1. **Set Explicit Versions**: Always specify `version` in `massdriver.yaml`
+   ```yaml
+   version: "1.2.3"  # Explicit semantic version
+   ```
+
+2. **Development Workflow**: Use development releases for testing
+   ```bash
+   # Set version in massdriver.yaml
+   version: "1.5.0"
+   
+   # Publish development release for testing
+   mass bundle publish --development-release  # Creates 1.5.0-dev.timestamp
+   # Outputs URL to view all test infrastructure and pipeline executions
+   
+   # Monitor plan/apply/compliance pipelines in real-time
+   
+   # Publish final release
+   mass bundle publish  # Creates 1.5.0
+   ```
+
+3. **Follow SemVer Guidelines**:
+   - **Patch** (`1.0.1`): Bug fixes, security updates, documentation
+   - **Minor** (`1.1.0`): New optional features, additional cloud regions
+   - **Major** (`2.0.0`): Breaking changes, removed parameters, incompatible updates
+
+4. **Version Progression**: Ensure each version is greater than the previous
+   - ✅ `1.0.0` → `1.0.1` → `1.1.0` → `2.0.0`
+   - ❌ Don't skip versions unnecessarily
+
+### For Platform Users
+
+1. **Production Environments**: Use exact versions for stability
+   ```
+   Set Version: "1.2.3"  # Pinned version for production
+   ```
+
+2. **Development/Staging**: Use "Latest" automation for newest features
+   ```
+   Set Version: "Latest"  # Auto-deploy to stable releases
+   ```
+
+3. **Release Channels**: Use tilde patterns for controlled automatic deployments
```

Comment:
Points 2 and 3 are roughly the same.
Point 2 is "use the latest release channel"
Point 3 is "use release channel"